nexus/src/app/background/tasks/instance_watcher.rs

```diff
@@ -44,9 +44,10 @@ pub(crate) struct InstanceWatcher {
     id: WatcherIdentity,
 }
 
-const MAX_SLED_AGENTS: NonZeroU32 = unsafe {
-    // Safety: last time I checked, 100 was greater than zero.
-    NonZeroU32::new_unchecked(100)
+// Chosen arbitrarily. Perhaps this should be configurable in the config file?
```

Comment:
I think I'd say a little bit more about this.  For example:

```suggestion
// Determines how many instance checks and recoveries are executed concurrently.  If this number is too high, there's risk that this task could starve other Nexus activities or overload the database or overload one or more sled agents.  The only consequence of the number being too low is that it takes longer for the system to notice that an instance is dead and needs to be dealt with, which translates to increased latency from a failure to the point where an instance gets restarted.  For now, we tune this pretty low, choosing safety over low recovery latency for this relatively rare event.
```

Reply:
Yeah, I think documenting the _motivation_ behind this is a good idea, thank you! However, I don't think the comment is _quite_ write as you've written it --- this task is not directly responsible for _restarting_ failed instances. It just triggers the instance-update sagas that reconcile the instance's state with its VMMs. So, the latency here is not the latency of restarting a failed instance, but the latency of updating the instance record's state to reflect changes in its VMM(s) states: either tearing down the resources associated with a `Destroyed` instance, moving an instance whose VMM has crashed to `Failed` so that it can be deleted/restarted, or updating an instance record to reflect a completed migration.

Reply:
Right, but isn't this a necessary step for the instances to be restarted?

Reply:
It is, yeah. I just didn't want the comment to suggest that that was _all_ this task was responsible for, or that it was the only source of latency for automatic restarts.

Reply:
Totally.  But I think if somebody is looking at this constant and wants to know the impact of changing it, they should know that in principle that's an important (maybe the only?) user-visible impact.

Reply:
Terrific.  I just noted a few copy edits.

Reply:
Thanks for catching my typos!